cmd/validators/main.go

```
@@ -120,12 +127,159 @@ type pubKeyInfo struct {
+					Type: "ed25519",
+					Data: pubkey,
+				},
+				Power: *powerFl,
+			},
+		)
+
+		raw, err := addValidatorTx.Marshal()
+		if err != nil {
+			return fmt.Errorf("cannot serialize transaction: %s", err)
+		}
+		out.Write(raw)
+		return nil
+	}
+}
+
+func cmdMultisigSign(in io.Reader, out io.Writer) func() error {
+	return func() error {
+		flag.Usage = func() {
+			fmt.Fprint(flag.CommandLine.Output(), `
+Sign given transaction. This is decoding a transaction data from standard
+input, adds a signature and writes back to standard output signed transaction
+content.
+
+`)
+			flag.PrintDefaults()
+		}
+		var (
+			tmAddrFl = flag.String("tm", "https://bns.NETWORK.iov.one:443", "Tendermint node address. Use proper NETWORK name.")
+			keyFl    = flag.String("key", "", "Hex encoded, private key that transaction should be signed with.")
+		)
+		flag.Parse()
+
+		if *keyFl == "" {
+			return errors.New("private key is required")
+		}
+		key, err := decodePrivateKey(*keyFl)
+		if err != nil {
+			return fmt.Errorf("cannot decode private key: %s", err)
+		}
+
+		raw, err := ioutil.ReadAll(in)
+		if err != nil {
+			return fmt.Errorf("cannot read transaction from stdin: %s", err)
+		}
+		if len(raw) == 0 {
+			return errors.New("no stdin data")
+		}
+		var tx app.Tx
+		if err := tx.Unmarshal(raw); err != nil {
+			return fmt.Errorf("cannot deserialize transaction: %s", err)
+		}
+
+		genesis, err := fetchGenesis(*tmAddrFl)
+		if err != nil {
+			return fmt.Errorf("cannot fetch genesis: %s", err)
+		}
+
+		bnsClient := client.NewClient(client.NewHTTPConnection(*tmAddrFl))
+		aNonce := client.NewNonce(bnsClient, key.PublicKey().Address())
+		if seq, err := aNonce.Next(); err != nil {
+			return fmt.Errorf("cannot get the next sequence number: %s", err)
+		} else {
+			client.SignTx(&tx, key, genesis.ChainID, seq)
+		}
+
+		if raw, err := tx.Marshal(); err != nil {
+			return fmt.Errorf("cannot serialize transaction: %s", err)
+		} else {
+			out.Write(raw)
```

pp: please return the err